runtime/stdlib/test.go

```diff
@@ -1517,7 +1527,7 @@ func newMatcherWithGenericTestFunction(
 
 			return value
 		},
-		matcherTestFunctionType,
+		matcherTestFieldType,
```

Comment:
Fix: Create a function that has the type of the `Matcher.test` field, `((AnyStruct): Bool)`, so the initialization of the field in `Matcher.init` works.